lib/Sema/LookupVisibleDecls.cpp

```diff
@@ -930,16 +932,35 @@ class OverrideFilteringConsumer : public VisibleDeclConsumer {
                           (BaseTy->getNominalOrBoundGenericNominal() ||
                            BaseTy->is<ArchetypeType>()) &&
                           VD->getDeclContext()->isTypeContext());
-      ModuleDecl *M = DC->getParentModule();
+
+      /// Substitue generic parameters in the signature of a found decl. The
+      /// returned type can be used to determine if we have already found a
+      /// conflicting declaration.
+      auto normalizeSignatureType = [&](CanType SignatureType, ValueDecl *VD,
```

Comment:
Maybe rename this to "substGenericArgs" since that's more descriptive than "normalize" which could mean anything